core/blockchain.go

```diff
@@ -1657,19 +1724,25 @@ func (self *BlockChain) InsertChain(chain types.Blocks) (chainIndex int, err err
 		stats.processed++
 	}
 
+	ev := ChainInsertEvent{
+		Processed: stats.processed,
+		Queued:    stats.queued,
+		Ignored:   stats.ignored,
+		TxCount:   txcount,
+	}
+	r := &ChainInsertResult{ChainInsertEvent: ev}
+	r.Index = 0 // NOTE/FIXME?(whilei): it's kind of strange that it returns 0 when no error... why not len(blocks)-1?
```

Comment:
This is just what seems to me kind of weird pattern that these Insert methods, after processing _n_ blocks, return an index _i = 0_ if there were not errors (and _i=errorN_ iff there was). It seems more intuitive to return _n=length-1_ when no errors. Not a big deal, just kind of backwards-seeming to me.